### PR TITLE
You can now re-initialize a Timber Image with a Timber Image

### PR DESCRIPTION
**Pull request by jarednova**
**Reviewer**: @connorjburton 

#### Issue
If the `TimberImage` or `Image` function in twig was passed a `\Timber\Image` object it would cause a compile error.

#### Solution
If passed an `Image` object, the `\Timber\Image::init` function will now grab the ID and re-initialize

#### Impact
It would be bad practice to do this, but it's worse to have a silent error. 

#### Usage
No.

#### Considerations
Rather than re-initialize, the `Image` object should be able to grab the public assignments of properties so that no new queries need to be run.

#### Testing
New Unit tests included demonstrating failing behavior and passing behavior

### CODE REVIEW

File: lib/Image.php [hunk 1]
```diff
@@ -212,7 +223,9 @@ public static function wp_upload_dir() {
 	 */
 	function init( $iid = false ) {
 		if ( !$iid ) { Helper::error_log('Initalized TimberImage without providing first parameter.'); return; }
-		
+		if ( $iid instanceof Image ) {
```
Comment (connorjburton) [inline lib/Image.php]: Would `$iid instanceof __CLASS__` work here? In case we ever change the class name.

**Comment by connorjburton** [comment]
Just that one note then I'll merge

File: lib/Image.php (hunk 1 above)
Comment (jarednova) [inline lib/Image.php]: You'd think that, right? but when I change it to...

```php
if ( $iid instanceof __CLASS__ )
```

I get: `Parse error: syntax error, unexpected '__CLASS__' (T_CLASS_C) in /srv/www/timber/lib/Image.php on line 226`

Comment (jarednova) [inline lib/Image.php]: ... though this works:

```php
if ( $iid instanceof self ) {
```

**Comment by jarednova** [comment]
@connorjburton check it out now!

**Comment by connorjburton** [comment]
Beautiful. Merged.

Thanks!

**Pull request merged**